app/controllers/publishers/site_banners_controller.rb

```diff
@@ -5,14 +5,104 @@ class Publishers::SiteBannersController < ApplicationController
 
   MAX_IMAGE_SIZE = 10_000_000
 
-  def new
-    @site_banner = current_publisher.site_banner || SiteBanner.new
+  #Fetch Banner by channel_id, if it does not exist create a template banner.
+  def fetch
+    site_banner = current_publisher.site_banners.where(channel_id: params[:channel_id]).first
+    if site_banner.present?
+      data = site_banner.read_only_react_property
+      data[:backgroundImage] = data[:backgroundUrl]
+      data[:logoImage] = data[:logoUrl]
+      render(json: data.to_json)
+    else
+      site_banner = SiteBanner.new
+      site_banner.update(
+        publisher_id: current_publisher.id,
+        channel_id: params[:channel_id],
+        title: 'Brave Rewards',
+        description:
+'Thanks for stopping by. We joined Brave\'s vision of protecting your privacy because we believe that fans like you would support us in our effort to keep the web a clean and safe place to be.
+
+Your tip is much appreciated and it encourages us to continue to improve our content.',
+        donation_amounts: [1, 5, 10],
+        default_donation: 5,
+        social_links: {'youtube': '', 'twitter': '', 'twitch': ''}
+      )
+      data = site_banner.read_only_react_property
+      data[:backgroundImage] = nil
+      data[:logoImage] = nil
+      render(json: data.to_json)
+    end
   end
 
-  def create
-    site_banner = current_publisher.site_banner || SiteBanner.new
+  def channels
+
+    # Handle edge case where user hasn't created any channels yet
+    if current_publisher.channels.length == 0
```

Comment:
nit: let's do `.zero?` instead of comparing with `== 0` :innocent:

Reply:
Done ✔️